src/MementoContainer/Analysis/PropertyAnalyzer.cs

```diff
@@ -128,11 +149,31 @@ public IList<IPropertyData> GetProperties(object obj)
         }
 
         /// <summary>
+        /// Tells if a given property has to be registered in the memento in case the class is decorated with MementoClassAttribute
+        /// </summary>
+        /// <param name="kv">Property info and its attributes</param>
+        /// <returns>True if the property must be registered in the memento</returns>
+        protected virtual bool PropertySelectorForMementoClass(KeyValuePair<PropertyInfo, IList<Attribute>> kv)
+        {
+            return kv.Value.Any(attr => attr is MementoCollectionAttribute) || kv.Key.PropertyType.IsCollection();
```

Comment:
Was this a copy-paste oversight? I think this is meant to be

```csharp
kv.Value.Any(attr => attr is MementoPropertyAttribute) || kv.Key.HasGetAndSet()
```

Reply:
Yes, probably I copied the wrong lines of code while organizing the code after making changes on a custom project I had prior to the fork.

Reply:
No worries, happens all the time :D